src/plugins/ResizeReorder/ReorderHandle.js

```diff
@@ -0,0 +1,241 @@
+        onTouchEnd={this.props.touchEnabled ? e => e.stopPropagation() : null}
+        onTouchMove={this.props.touchEnabled ? e => e.stopPropagation() : null}
+        style={style}>
+      </div>
+    );
+  }
+
+  /**
+   *
+   * @param {MouseEvent} event
+   */
+  onMouseDown = (event) => {
+    this.distance = 0;
+    this.animating = true;
+    this.scrollStart = getState().scrollX;
+    this.originalLeft = this.props.left;
+    this.initializeDOMMouseMoveTracker();
+    this.mouseMoveTracker.captureMouseMoves(event);
+    this.updateParentReorderingData({
+      isColumnReordering: true,
+      displacement: 0
+    });
+    this.frameId = requestAnimationFrame(this.updateState);
+  };
+
+  /**
+   * @param {number} deltaX
+   */
+  onMouseMove = (deltaX) => {
+    this.distance += deltaX * (this.props.isRTL ? -1 : 1);
+  };
+
+  onMouseUp = () => {
+    cancelAnimationFrame(this.frameId);
+    this.calculateColumnOrder();
+    this.animating = false;
+    this.frameId = null;
+    this.distance = 0;
+    this.mouseMoveTracker.releaseMouseMoves();
+    this.updateParentReorderingData({
+      isColumnReordering: false,
+      displacement: 0
+    });
+  };
+
+  initializeDOMMouseMoveTracker = () => {
+    this.mouseMoveTracker = new DOMMouseMoveTracker(
+      this.onMouseMove,
+      this.onMouseUp,
+      document.body,
+      this.props.touchEnabled
+    );
+  };
+
+  updateParentReorderingData = (state) => {
+    this.props.updateParentReorderingData(state);
+  };
+
+  updateState = () => {
+    if (this.animating) {
+      this.frameId = requestAnimationFrame(this.updateState);
+    }
+    this.calculateDisplacementWithScroll();
+  };
+
+  calculateDisplacementWithScroll = () => {
+    const scrollStart = this.scrollStart;
+    let { isFixed, maxScrollX } = this.props;
+    let { scrollX } = getState();
+    let deltaX = this.distance;
+    if (!isFixed) {
+      // Relative dragX position on scroll
+      const dragX = this.originalLeft - scrollStart + deltaX;
+      const availableScrollWidth = this.props.availableScrollWidth;
+      deltaX += scrollX - scrollStart;
+
+      // Scroll the table left or right if we drag near the edges of the table
+      if (dragX > availableScrollWidth - DRAG_SCROLL_BUFFER) {
+        scrollX = Math.min(scrollX + DRAG_SCROLL_SPEED, maxScrollX);
+      } else if (dragX <= DRAG_SCROLL_BUFFER) {
+        scrollX = Math.max(scrollX - DRAG_SCROLL_SPEED, 0);
+      }
+      this.props.scrollToX(scrollX);
+    }
+    this.updateParentReorderingData({ displacement: deltaX });
+  };
+
+  /**
+   * Returns true if column moved to right
+   * @param {number} deltaX
+   * @returns {boolean}
+   */
+  isColumnMovedToRight = (deltaX) => deltaX > 0;
+
+  /**
+   * Returns true if column moved to left
+   * @param {number} deltaX
+   * @returns {boolean}
+   */
+  isColumnMovedToLeft = (deltaX) => deltaX < 0;
+
+  calculateColumnOrder = () => {
```

Comment:
Let's have this function pure - i.e. just calculate the column indexes and return them. The parent can then use it to call the callback prop. You might also want to rename the function.